contracts/interchain-token-service/src/contract.rs

```diff
@@ -599,10 +609,71 @@ impl InterchainTokenService {
         env.crypto().keccak256(&chain_name.to_xdr(env)).into()
     }
 
-    fn canonical_token_deploy_salt(env: &Env, token_address: Address) -> BytesN<32> {
-        let chain_name_hash = Self::chain_name_hash(env);
-        env.crypto()
-            .keccak256(&(PREFIX_CANONICAL_TOKEN_SALT, chain_name_hash, token_address).to_xdr(env))
-            .into()
+    /// Deploys a remote token on a specified destination chain.
+    ///
+    /// This function authorizes the caller, retrieves the token's metadata,
+    /// validates the metadata, and emits an event indicating the start of the
+    /// token deployment process. It also constructs and sends the deployment
+    /// message to the remote chain.
+    ///
+    /// # Arguments
+    /// * `env` - Reference to the environment object.
+    /// * `caller` - Address of the caller initiating the deployment.
+    /// * `deploy_salt` - Unique salt used for token deployment.
+    /// * `destination_chain` - The name of the destination chain where the token will be deployed.
+    /// * `gas_token` - The token used to pay for gas during the deployment.
+    ///
+    /// # Returns
+    /// Returns the token ID of the deployed token on the remote chain, or an error if the deployment fails.
+    ///
+    /// # Errors
+    /// Returns `ContractError` if the deployment fails, the token ID is invalid, or token metadata is invalid.
+    fn deploy_remote_token(
+        env: &Env,
+        caller: Address,
+        deploy_salt: BytesN<32>,
+        destination_chain: String,
+        gas_token: Token,
+    ) -> Result<BytesN<32>, ContractError> {
+        caller.require_auth();
+
+        let token_id = Self::interchain_token_id(env, Address::zero(env), deploy_salt);
+        let token_address = env
+            .storage()
+            .persistent()
+            .get::<_, TokenIdConfigValue>(&DataKey::TokenIdConfigKey(token_id.clone()))
+            .ok_or(ContractError::InvalidTokenId)?
+            .token_address;
+        let token = token::Client::new(env, &token_address);
+        let token_metadata = TokenMetadata {
+            name: token.name(),
+            decimal: token.decimals(),
+            symbol: token.symbol(),
+        };
+
+        assert_ok!(validate_token_metadata(token_metadata.clone()));
```

Comment:
assert_ok is still being used in contract code

Reply:
Ah, I updated the code, but there were conflicts, I rebased the code and then the change was gone. It’s now updated.